### PR TITLE
add basic nightly runner framework

### PR DESCRIPTION
**Pull request by ewlu**

### CODE REVIEW

File: .github/workflows/downtime-runner.yaml [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+          sudo apt remove python3-pip
+          wget https://bootstrap.pypa.io/get-pip.py
+          sudo python3 get-pip.py
+          pip install -U pyopenssl cryptography
+
+      - name: Get Timestamps
+        run: |
+          TZ=UTC date -d "$(date -d 'yesterday' '+%Y-%m-%d')" "+%Y-%m-%dT%H:%M:%S" > yesterday_start_date.txt
+          END_TIME=$(cat yesterday__start_date.txt)
+          TZ=UTC date -d "@$((($(TZ=UTC date -d $END_TIME +%s) + 86399)))" '+%Y-%m-%dT%H:%M:%S' > yesterday_end_date.txt
```
Comment (patrick-rivos) [inline .github/workflows/downtime-runner.yaml:107]:
```suggestion
          # One second less than 24 hrs
          TZ=UTC date -d "@$((($(TZ=UTC date -d $END_TIME +%s) + 86399)))" '+%Y-%m-%dT%H:%M:%S' > yesterday_end_date.txt
```

File: .github/workflows/downtime-runner.yaml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+      - name: Bump pyopenssl and crypto
+        run: |
+          sudo apt remove python3-pip
+          wget https://bootstrap.pypa.io/get-pip.py
+          sudo python3 get-pip.py
+          pip install -U pyopenssl cryptography
+
+      - name: Get previous run id
+        run: |
+          python scripts/get_previous_timestamp.py -token ${{ secrets.GITHUB_TOKEN }} -rid ${{ github.run_id }} -repo ewlu/gcc-precommit-ci -workflow Downtime-Runner
```
Comment (patrick-rivos) [inline .github/workflows/downtime-runner.yaml:38]: This only checks if patchworks is down at this moment? Shouldn't we check if it's been down at any point via artifact shenanigans?

**Review by patrick-rivos — CHANGES_REQUESTED** [review]

File: .github/workflows/downtime-runner.yaml (hunk 2 above)
Comment (ewlu) [reply .github/workflows/downtime-runner.yaml:38]: It's checked 3 steps down. Not sure if that's the best way but open to suggestions

File: .github/workflows/downtime-runner.yaml [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    defaults:
+      run:
+        working-directory: riscv-gnu-toolchain
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Clone riscv-gnu-toolchain
+        working-directory: ./
+        run: |
+          git submodule update --init riscv-gnu-toolchain
```
Comment (patrick-rivos) [inline .github/workflows/downtime-runner.yaml:21]: Try deleting this if we only need access to /scripts - if that doesn't work then use the cache for this

File: .github/workflows/downtime-runner.yaml [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
```
Comment (patrick-rivos) [inline .github/workflows/downtime-runner.yaml:85]: Ditto

File: .github/workflows/patchworks.yaml [hunk 5]
```diff
@@ -82,6 +82,9 @@ jobs:
 
       - name: Get list of new patches
         run: |
+          if test -f "./artifact_names.txt"; then
```
Comment (patrick-rivos) [inline .github/workflows/patchworks.yaml:85]: nit: use the bracket notation for test

**Review by patrick-rivos — APPROVED** [review]

**Pull request merged**